spec/acceptance/5_client_spec.rb

```diff
@@ -35,6 +35,24 @@ class { 'keycloak':
         service_accounts_enabled       => true,
         browser_flow                   => 'foo',
       }
+      keycloak_client { 'saml.foo.bar':
+        realm                                   => 'test',
+        root_url                                => 'https://saml.foo.bar/root',
+        admin_url                               => 'https://saml.foo.bar/admin',
+        base_url                                => 'https://saml.foo.bar',
+        redirect_uris                           => ['https://saml.foo.bar/test1'],
+        default_client_scopes                   => ['address'],
```

Comment:
This `default_client_scopes` is being re-applied and so not idempotent.  On my Keycloak install the `address` scope is specific to openid-connect so likely can't be used for SAML client.  The only SAML default scope I see on my isntalls is `role_list`, the rest are ones I've created.  So maybe either remove `default_client_scopes` or change the value to `['role_list']`.

Reply:
thank you for your hints. I am sorry to waste your time with such mistakes, but as I have written, I am no puppet developer, so I have really zero experience.

I have used `role_list` and I think, I should define the check of new saml client too. But now it fails, I try to find out the reason and fix it.